Trim URL at `HomeView`

```
Caused by: java.net.URISyntaxException: Illegal character in authority at index 7: http:// quay.io/kyberorg/yalsee-base:distroless-java,
	... 96 common frames omitted,
java.lang.RuntimeException: String ' quay.io/kyberorg/yalsee-base:distroless-java': malformed URL or not URL at all,
	at io.kyberorg.yalsee.utils.UrlUtils.makeFullUri(UrlUtils.java:71) ~[classes/:edge],
	at io.kyberorg.yalsee.ui.HomeView.onSaveLink(HomeView.java:315) ~[classes/:edge]
```

--- a/src/main/java/io/kyberorg/yalsee/ui/HomeView.java
+++ b/src/main/java/io/kyberorg/yalsee/ui/HomeView.java
@@ -312,7 +312,7 @@ public class HomeView extends HorizontalLayout {
             isFormValid = false;
         } else {
             try {
-                longUrl = UrlUtils.makeFullUri(longUrl).toString();
+                longUrl = UrlUtils.makeFullUri(longUrl.trim()).toString();
             } catch (RuntimeException e) {
                 log.error("{} URL validation failed", TAG);
                 log.debug("", e);

--- a/src/test/java/io/kyberorg/yalsee/test/pageobjects/external/QuayIo.java
+++ b/src/test/java/io/kyberorg/yalsee/test/pageobjects/external/QuayIo.java
@@ -1,0 +1,18 @@
+package io.kyberorg.yalsee.test.pageobjects.external;
+
+import com.codeborne.selenide.SelenideElement;
+
+import static com.codeborne.selenide.Selenide.$;
+
+/**
+ * Page object for page https://quay.io .
+ *
+ * @since 3.4
+ */
+public final class QuayIo {
+    public static final SelenideElement LOGO = $("#quay-logo");
+
+    private QuayIo() {
+        throw new UnsupportedOperationException("Utility class");
+    }
+}

--- a/src/test/java/io/kyberorg/yalsee/test/ui/home/HomePageTest.java
+++ b/src/test/java/io/kyberorg/yalsee/test/ui/home/HomePageTest.java
@@ -5,6 +5,7 @@ import io.kyberorg.yalsee.test.pageobjects.HomePageObject;
 import io.kyberorg.yalsee.test.pageobjects.MyLinksViewPageObject;
 import io.kyberorg.yalsee.test.pageobjects.NotFoundViewPageObject;
 import io.kyberorg.yalsee.test.pageobjects.RedirectPageObject;
+import io.kyberorg.yalsee.test.pageobjects.external.QuayIo;
 import io.kyberorg.yalsee.test.pageobjects.external.VR;
 import io.kyberorg.yalsee.test.ui.SelenideTest;
 import io.kyberorg.yalsee.test.utils.SelenideUtils;
@@ -213,8 +214,24 @@ public class HomePageTest extends SelenideTest {
         HomePageObject.MainArea.DESCRIPTION_INPUT.shouldBe(empty);
     }
 
+    public void linkWithSpacesShouldBeTrimmedAndSaved() {
+        HomePageObject.pasteValueInFormAndSubmitIt(" quay.io/kyberorg/yalsee-base");
+
+        SelenideElement shortLink = HomePageObject.ResultArea.RESULT_LINK;
+        $(shortLink).shouldBe(visible);
+        String shortUrl = shortLink.getText();
+
+        open(shortUrl + addRedirectPageBypassSymbol());
+        verifyThatQuayIoOpened();
+    }
+
     private void verifyThatVROpened() {
         Assertions.assertEquals(VR.TITLE_TEXT, SelenideUtils.getPageTitle());
+    }
+
+    private void verifyThatQuayIoOpened() {
+        QuayIo.LOGO.should(exist);
+        QuayIo.LOGO.shouldBe(visible);
     }
 
     private void verifyThatRedirectPageOpened() {

--- a/src/test/java/io/kyberorg/yalsee/test/ui/home/HomePageTest.java
+++ b/src/test/java/io/kyberorg/yalsee/test/ui/home/HomePageTest.java
@@ -214,6 +214,10 @@ public class HomePageTest extends SelenideTest {
         HomePageObject.MainArea.DESCRIPTION_INPUT.shouldBe(empty);
     }
 
+    /**
+     * Tests that link with spaces is trimmed and saved.
+     */
+    @Test
     public void linkWithSpacesShouldBeTrimmedAndSaved() {
         HomePageObject.pasteValueInFormAndSubmitIt(" quay.io/kyberorg/yalsee-base");
 

